From d4a8dfac50acf546be29fd8886868b550a4f9dbc Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Thu, 5 Oct 2017 15:51:50 +0530
Subject: [PATCH] Skip unshare() and capset() tests on Linux

These operations require certain capabilities which are most likely not
available in the CI systems like CircleCI
---
 src/syscall/exec_linux_test.go | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/src/syscall/exec_linux_test.go b/src/syscall/exec_linux_test.go
index 114deec..2ae0c3a 100644
--- a/src/syscall/exec_linux_test.go
+++ b/src/syscall/exec_linux_test.go
@@ -75,6 +75,8 @@ func checkUserNS(t *testing.T) {
 }
 
 func whoamiCmd(t *testing.T, uid, gid int, setgroups bool) *exec.Cmd {
+        t.Skip("skipping test on CI Systems; CLONE_NEWUSER requires CAP_SYS_ADMIN")
+
 	checkUserNS(t)
 	cmd := exec.Command("whoami")
 	cmd.SysProcAttr = &syscall.SysProcAttr{
@@ -237,6 +239,8 @@ func TestGroupCleanup(t *testing.T) {
 }
 
 func TestGroupCleanupUserNamespace(t *testing.T) {
+        t.Skip("skipping test on CI Systems; CLONE_NEWUSER requires CAP_SYS_ADMIN")
+
 	if os.Getuid() != 0 {
 		t.Skip("we need root for credential")
 	}
@@ -318,7 +322,8 @@ func TestUnshareMountNameSpace(t *testing.T) {
 
 	o, err := cmd.CombinedOutput()
 	if err != nil {
-		if strings.Contains(err.Error(), ": permission denied") {
+		if strings.Contains(err.Error(), ": permission denied") ||
+                   strings.Contains(err.Error(), ": operation not permitted") {
 			t.Skipf("Skipping test (golang.org/issue/19698); unshare failed due to permissions: %s, %v", o, err)
 		}
 		t.Fatalf("unshare failed: %s, %v", o, err)
@@ -376,7 +381,8 @@ func TestUnshareMountNameSpaceChroot(t *testing.T) {
 
 	o, err := cmd.CombinedOutput()
 	if err != nil {
-		if strings.Contains(err.Error(), ": permission denied") {
+		if strings.Contains(err.Error(), ": permission denied") ||
+                   strings.Contains(err.Error(), ": operation not permitted") {
 			t.Skipf("Skipping test (golang.org/issue/19698); unshare failed due to permissions: %s, %v", o, err)
 		}
 		t.Fatalf("unshare failed: %s, %v", o, err)
@@ -477,6 +483,8 @@ func TestAmbientCapsHelper(*testing.T) {
 }
 
 func TestAmbientCaps(t *testing.T) {
+        t.Skip("skipping test on CI Systems; capset() probably requires CAP_SETPCAP")
+
 	// Make sure we are running as root so we have permissions to use unshare
 	// and create a network namespace.
 	if os.Getuid() != 0 {
-- 
2.7.4

